azext_iot/common/certops.py

```diff
@@ -80,4 +81,4 @@ def open_certificate(certificate_path):
                 certificate = certificate.decode("utf-8")
             except UnicodeError:
                 certificate = base64.b64encode(certificate).decode("utf-8")
-    return certificate
+    return certificate.rstrip()  # Remove trailing white space from the certificate content
```

Comment:
This is added back in as the service seems to not able to deal with cert ended with an empty new line (again?). It does no harm to us and it more robust.